src/compiler/crystal/macros.cr

```diff
@@ -794,6 +798,10 @@ module Crystal::Macros
 
   # A named tuple literal.
   class NamedTupleLiteral < ASTNode
+    # Similar to `Hash#delete`.
```

Comment:
I think is similar to `Hash#delete` but one thing is this will change the Type signature.  Should that be added as a warning to the docs?

Reply:
It doesn't have a type technically since this is in macro land.